edwin/client/static/js/actions/TimelineActions.js

```diff
@@ -115,9 +121,36 @@ export function loadCommentTags(bugIds) {
       type: TimelineConstants.ActionTypes.SET_COMMENT_TAGS,
       commentSpecs,
     });
-  });
+  })
+  .then(() => ProgressActions.endTask('Load comments'));
 }
 
+/**
+ * Fetch data for blocker bugs.
+ * @param {Array<Immutable.List>} List of bug ids
+ */
+export function loadBlockerBugs(bugIds) {
+  if (bugIds.size > 0) {
+    let bugQuery = {id: Array.from(bugIds).join(',')};
+    const user = UserStore.getAll();
+    ProgressActions.startTask('Load blockers');
+
+    if (user.get('loggedIn')) {
+      query.api_key = user.get('apiKey');
+    }
+
+    return bzAPI.getBugs(bugQuery)
+    .then(newBugs => {
+      Dispatcher.dispatch({
+        type: TimelineConstants.ActionTypes.SET_BLOCKER_BUGS,
+        newBugs,
+      });
+    })
+    .then(() => ProgressActions.endTask('Load blockers'));
+  } else {
+    return undefined;
```

Comment:
Uh oh. It probably isn't good to sometimes return a Promise and sometimes not. Maybe return something like `Promise.resolve()` (return a promise that is resolved to `undefined`) instead?